src/xrpld/app/tx/detail/LoanSet.cpp

```diff
@@ -88,10 +88,12 @@ LoanSet::preflight(PreflightContext const& ctx)
     if (auto const paymentInterval = tx[~sfPaymentInterval];
         !validNumericMinimum(paymentInterval, LoanSet::minPaymentInterval))
         return temINVALID;
-
-    else if (!validNumericRange(
-                 tx[~sfGracePeriod],
-                 paymentInterval.value_or(LoanSet::defaultPaymentInterval)))
+    // Grace period is between min default value and payment interval
+    else if (auto const gracePeriod = tx[~sfGracePeriod];  //
+             !validNumericRange(
+                 gracePeriod,
+                 paymentInterval.value_or(LoanSet::defaultPaymentInterval),
+                 defaultGracePeriod))
```

Comment:
I don't think it's documented anywhere in the spec that the grace period has a minimum value. It has a _default_, but a default is not automatically a minimum, unless I missed something.